components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2034,9 +2043,18 @@ public ServiceProvider getApplicationWithRequiredAttributes(int applicationId, L
         try {
             // Load basic application data
             ServiceProvider serviceProvider = getBasicApplicationData(applicationId, connection);
-            if (CollectionUtils.isNotEmpty(requiredAttributes)) {
+            int tenantID = IdentityTenantUtil.getTenantId(serviceProvider.getTenantDomain());
+
+            // Avoid iterating over similar items in requiredAttributes list.
+            Set<String> requiredAttributesSet = new HashSet<>(requiredAttributes);
+            if (requiredAttributesSet.contains(CLIENT_ID_SP_PROPERTY_NAME) &&
```

Comment:
why do we need to check for `CLIENT_ID_SP_PROPERTY_NAME` here?

Reply:
The logic here is to remove one if both are there in the requiredAttributesSet variable.